### ISSUE TITLE
Clicking table headers causes high CPU consumption

### ISSUE THREAD

**Issue by 3a9LL** [fyne-io/fyne#4264]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

When I click on table headers a few times I get the app hagned. CPU usage above 12%

### How to reproduce

1. Run Fyne Demo
2. Go to Collections -> Table
3. Click once on table headers
4. In a moment the app doesn't respond

or run the sample app below and do the steps since the 3rd

### Screenshots

![image](https://github.com/fyne-io/fyne/assets/7979983/c62809ae-1f92-4063-88ae-5b8f676ca765)
![image](https://github.com/fyne-io/fyne/assets/7979983/4e03a940-128a-42e5-95ee-f8d48165cbab)

### Example code

```go
package main

import (
	"fyne.io/fyne/v2"
	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/container"
	"fyne.io/fyne/v2/data/binding"
	"fyne.io/fyne/v2/widget"
	"os"
)

func main() {
	os.Setenv("FYNE_THEME", "light")

	myApp := app.New()
	w := myApp.NewWindow("Table with headers")

	tab := makeTableTab()

	tabs := container.NewAppTabs(
		container.NewTabItem("Table", tab),
	)
	tabs.SetTabLocation(container.TabLocationLeading)
	content := container.NewBorder(nil, nil, nil, nil, tabs)
	w.SetContent(content)
	w.Resize(fyne.NewSize(500, 300))

	w.ShowAndRun()

}

type TestStruct struct {
	Name        string
	Path        string
	Description string
}

func makeTableTab() fyne.CanvasObject {
	list := binding.NewUntypedList()
	table := widget.NewTableWithHeaders(
		func() (int, int) {
			return list.Length(), 3
		},
		func() fyne.CanvasObject {
			return widget.NewLabel("template")
		},
		func(id widget.TableCellID, o fyne.CanvasObject) {
			label := o.(*widget.Label)
			ti, _ := list.GetValue(id.Row)
			t := ti.(TestStruct)
			switch id.Col {
			case 0:
				label.SetText(t.Name)
			case 1:
				label.SetText(t.Path)
			case 2:
				label.SetText(t.Description)
			}
		})
	table.CreateHeader = func() fyne.CanvasObject {
		return widget.NewLabel("template")
	}
	colNames := []string{"Name", "Path", "Description"}
	table.UpdateHeader = func(i widget.TableCellID, t fyne.CanvasObject) {
		l := t.(*widget.Label)
		if i.Row == -1 {
			l.SetText(colNames[i.Col])
		}
	}
	table.SetColumnWidth(0, 100)
	table.SetColumnWidth(1, 100)
	table.SetColumnWidth(2, 300)
	table.ShowHeaderColumn = false

	return table
}

```

### Fyne version

2.4.0

### Go compiler version

1.21.0

### Operating system and version

Microsoft Windows [Version 10.0.19044.3086]

### Additional Information

![image](https://github.com/fyne-io/fyne/assets/7979983/0518c2a0-eecf-4f4a-b503-4d7c63a8ca99)

**Comment by andydotxyz** [comment]
Sounds like a duplicate of #4210

**Comment by 3a9LL** [comment]
I clicked 3-5 times and the app didn't respond. If I click on table cells I don't have the issue

**Comment by andydotxyz** [comment]
This is very strange, as table headers are not Tappable!

**Comment by 3a9LL** [comment]
Actually I clicked once on headers and the app hanged. Can anyone reproduce it?

**Comment by 3a9LL** [comment]
![image](https://github.com/fyne-io/fyne/assets/7979983/8e833cbc-650a-4e4b-96b3-497679a7eff5)

**Comment by GaetanBaert** [comment]
I got the same trouble here. The app froze when I click on the Header.

**Comment by andydotxyz** [comment]
This is now resolved on develop and will go into v2.4.1
